test/common_test/joint_features.cc

```diff
@@ -2550,6 +2550,153 @@ TEST_F(FixedJointFreeGroupFeatureTestTypes, FixedJointFreeGroupMove)
   }
 }
 
+using FixedJointWeldFeatureList = gz::physics::FeatureList<
+    gz::physics::FindFreeGroupFeature,
+    gz::physics::SetFreeGroupWorldPose,
+    gz::physics::AttachFixedJointFeature,
+    gz::physics::DetachJointFeature,
+    gz::physics::ForwardStep,
+    gz::physics::GetBasicJointProperties,
+    gz::physics::GetBasicJointState,
+    gz::physics::GetEngineInfo,
+    gz::physics::GetJointFromModel,
+    gz::physics::GetLinkFromModel,
+    gz::physics::GetModelFromWorld,
+    gz::physics::LinkFrameSemantics,
+    gz::physics::SetBasicJointState,
+    gz::physics::SetFixedJointWeldChildToParentFeature,
+    gz::physics::SetJointTransformFromParentFeature,
+    gz::physics::SetJointVelocityCommandFeature,
+    gz::physics::sdf::ConstructSdfModel,
+    gz::physics::sdf::ConstructSdfWorld
+>;
+
+using FixedJointWeldFeatureTestTypes =
+  JointFeaturesTest<FixedJointWeldFeatureList>;
+
+TEST_F(FixedJointWeldFeatureTestTypes, FixedJointWeldFall)
+{
+  // Attach joint between links from 2 models (M1 and M2) with a fixed joint.
+  // Let them fall and the child model (M2) should collide with a third model
+  // (body).
+  // Run the test once without welding child to parent then run again
+  // with welding enabled. Verify that the fixed joint enforces the relative
+  // pose between the parent and child links but there should still be some
+  // position error. Verify that the welded joint has a smaller error.
+  for (const std::string &name : this->pluginNames)
+  {
+    std::cout << "Testing plugin: " << name << std::endl;
+    gz::plugin::PluginPtr plugin = this->loader.Instantiate(name);
+
+    auto engine =
+        gz::physics::RequestEngine3d<FixedJointWeldFeatureList>::From(
+        plugin);
+    ASSERT_NE(nullptr, engine);
+
+    sdf::Root root;
+    const sdf::Errors errors = root.Load(common_test::worlds::kJointAcrossModelsSdf);
+    ASSERT_TRUE(errors.empty()) << errors.front();
+
+    auto world = engine->ConstructWorld(*root.WorldByIndex(0));
+
+    const std::string modelName1{"M1"};
+    const std::string modelName2{"M2"};
+    const std::string bodyName{"body"};
+
+    auto model1 = world->GetModel(modelName1);
+    auto model2 = world->GetModel(modelName2);
+    auto body = world->GetModel(bodyName);
+    auto model1Body = model1->GetLink(bodyName);
+    auto model2Body = model2->GetLink(bodyName);
+
+    const gz::math::Pose3d model1Pose(0, -1, 1.0, 0, 0, 0.0);
+    const gz::math::Pose3d model2Pose(0, 1, 1.0, 0, 0, 0.0);
+    const gz::math::Pose3d bodyPose(0, 1, 0.25, 0, 0, 0.0);
```

Comment:
these test poses don't have any rotations, so it doesn't confirm that the pose rotations are calculated correctly. Consider adding a todo comment to use more complex rotational values in the test

Reply:
good idea, added todo note. [74f4d61](https://github.com/gazebosim/gz-physics/pull/759/commits/74f4d61aba47213f89ebfbb180adf3d32c5b1cf9)